tx.go

```diff
@@ -67,3 +74,32 @@ func GetPath(tx Tx) string {
 
 // TxDecoder can parse bytes into a Tx
 type TxDecoder func(txBytes []byte) (Tx, error)
+
+// ExtractMsgFromSum will find a weave message from a tx sum type if it exists.
+// Assuming you define your Tx with protobuf, this will help you implement GetMsg()
+//
+//   ExtractMsgFromSum(tx.GetSum())
+//
+// To work, this requires sum to be a pointer to a struct with one field,
+// and that field can be cast to a Msg.
+// Returns an error if it cannot succeed.
+func ExtractMsgFromSum(sum interface{}) (Msg, error) {
+	// TODO: add better error messages here with new refactor
+	if sum == nil {
+		return nil, errors.ErrInternal("sum is <nil>")
```

Comment:
Sum bothers me in general, it does not self document. As this is basically transaction value aka it's concrete type. Might not be a concern for this PR, just something that I noticed. Seeing "sum is <nil>" in the log is not very descriptive, but at the same time this error should never happen. So let's leave it at that, but I'd think about changing this naming to something more self-documenting.